Tasks/Common/webdeployment-common-v2/ziputility.ts

```diff
@@ -2,13 +2,87 @@ import tl = require('azure-pipelines-task-lib/task');
 import path = require('path');
 import Q = require('q');
 import fs = require('fs');
-import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+import tr = require('azure-pipelines-task-lib/toolrunner');
 
 var DecompressZip = require('decompress-zip');
 var archiver = require('archiver');
 
 const deleteDir = (path: string) => tl.exist(path) && tl.rmRF(path);
 
+const extractWindowsZip = async (fromFile: string, toDir: string, usePowerShell?: boolean) => {
+    if (usePowerShell) {
+        await extractUsingPowerShell(fromFile, toDir);
+    }
+    else {
+        await extractUsing7zip(fromFile, toDir);
+    }
+}
+
+const extractUsingPowerShell = async (fromFile: string, toDir: string) => {
+    tl.debug(`Using PowerShell for extracting zip ${fromFile}`);
+    let command = `Expand-Archive -Path "${fromFile}" -DestinationPath "${toDir}"`;
```

Comment:
Consider adding `-Force` to make Expand-Archive to create the destination path directory if it does not exists already as well as to not show ask for any confirmation.

Reply:
done.